### ISSUE TITLE
crash on ctrl-c/SIGINT during `terraform init` when supplying values on the cli

### ISSUE THREAD

**Issue by mildwonkey** [hashicorp/terraform#22167]
### Terraform Version
current from master: 
```
$ terraform -v
Terraform v0.12.5-25-g16fa18c13 (16fa18c139b5460b81b09779ee57dab0bb661949)2
```
### Terraform Configuration Files
Note the commented-out organization. Terraform requests the missing attributes on the cli after running `terraform init`.
```hcl
terraform {
  backend "remote" {
    hostname = "app.terraform.io"
    //organization = "mildwonkey"
    workspaces {
      name = "example"
    }
  }
}
```

### Crash Output
```
$ terraform init

Initializing the backend...
Backend configuration changed!

Terraform has detected that the configuration specified for the backend
has changed. Terraform will now check for existing state in the backends.


organization
  The name of the organization containing the targeted workspace(s).

  Enter a value: ^C
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]:
github.com/hashicorp/terraform/vendor/github.com/zclconf/go-cty/cty.Value.GetAttr(0x2e117e0, 0xc000010550, 0x0, 0x0, 0x299c4e4, 0xc, 0x100d839, 0x2e111e0, 0x10, 0x10)
	/Users/kristin/go/src/github.com/hashicorp/terraform/vendor/github.com/zclconf/go-cty/cty/value_ops.go:620 +0x28c
github.com/hashicorp/terraform/backend/remote.(*Remote).PrepareConfig(0xc0005a9970, 0x2e117e0, 0xc000010550, 0x0, 0x0, 0x1, 0xc000077760, 0x1, 0x1, 0x403e640, ...)
	/Users/kristin/go/src/github.com/hashicorp/terraform/backend/remote/backend.go:146 +0x7c
github.com/hashicorp/terraform/command.(*Meta).backendInitFromConfig(0xc0000ef520, 0xc00028d180, 0xc7, 0xc000300270, 0xc7, 0xc000300000, 0xce, 0xc0000d0060, 0xc0001100e0, 0x46, ...)
	/Users/kristin/go/src/github.com/hashicorp/terraform/command/meta_backend.go:1046 +0x2df
github.com/hashicorp/terraform/command.(*Meta).backend_C_r_S_changed(0xc0000ef520, 0xc00028d180, 0x28915efd, 0xc0000d08a0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/kristin/go/src/github.com/hashicorp/terraform/command/meta_backend.go:806 +0xa2
github.com/hashicorp/terraform/command.(*Meta).backendFromConfig(0xc0000ef520, 0xc000201ea0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/kristin/go/src/github.com/hashicorp/terraform/command/meta_backend.go:550 +0x9a7
github.com/hashicorp/terraform/command.(*Meta).Backend(0xc0000ef520, 0xc000201ea0, 0x0, 0xc000547800, 0x1, 0x1, 0x0)
	/Users/kristin/go/src/github.com/hashicorp/terraform/command/meta_backend.go:85 +0x5ee
github.com/hashicorp/terraform/command.(*InitCommand).initBackend(0xc0000ef520, 0xc0002da460, 0x299fa40, 0xf, 0xc00000f9a0, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/kristin/go/src/github.com/hashicorp/terraform/command/init.go:449 +0x1a9
github.com/hashicorp/terraform/command.(*InitCommand).Run(0xc0000ef520, 0xc0000d2010, 0x0, 0x0, 0xc000162680)
	/Users/kristin/go/src/github.com/hashicorp/terraform/command/init.go:286 +0x1925
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc0003ca000, 0xc0003ca000, 0xc000547d90, 0x1)
	/Users/kristin/go/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:255 +0x1f1
main.wrappedMain(0x0)
	/Users/kristin/go/src/github.com/hashicorp/terraform/main.go:223 +0xaff
main.realMain(0x0)
	/Users/kristin/go/src/github.com/hashicorp/terraform/main.go:100 +0xb4
main.main()
	/Users/kristin/go/src/github.com/hashicorp/terraform/main.go:36 +0x3b
```

### Steps to Reproduce
1. create a .tf file with a backend config which is missing a required field 
1. `terraform init`
1. Hit `ctrl-c` when terraform prompts for this missing value
1. bang! crash! the lightning flashed!  

### Related Issues
https://github.com/hashicorp/terraform/pull/20546

**Comment by aditsachde** [comment]
Hit the exact same issue

**Crash output**

```
2019/08/24 12:56:54 [INFO] Terraform version: 0.12.6  
2019/08/24 12:56:54 [INFO] Go runtime version: go1.12.4
2019/08/24 12:56:54 [INFO] CLI args: []string{"/home/adit/bin/terraform", "init", "-backend-config=/proc/self/fd/18"}
2019/08/24 12:56:54 [DEBUG] Attempting to open CLI config file: /home/adit/.terraformrc
2019/08/24 12:56:54 Loading CLI configuration from /home/adit/.terraformrc
2019/08/24 12:56:54 [INFO] CLI command args: []string{"init", "-backend-config=/proc/self/fd/18"}
2019/08/24 12:56:54 [TRACE] Meta.Backend: merging -backend-config=... CLI overrides into backend configuration
2019/08/24 12:56:54 [TRACE] Meta.Backend: built configuration for "remote" backend with hash value 4115669078
2019/08/24 12:56:54 [TRACE] Meta.Backend: backend has not previously been initialized in this working directory
2019/08/24 12:56:54 [DEBUG] New state was assigned lineage "dbb0962c-2b4c-93d7-af30-250186005d11"
2019/08/24 12:56:54 [TRACE] Meta.Backend: moving from default local state only to "remote" backend
2019/08/24 12:56:54 [DEBUG] command: asking for input: "organization"
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.GetAttr(0x2228c40, 0xc00000e940, 0x0, 0x0, 0x1dadead, 0xc, 0x40b849, 0x2228a80, 0x10, 0x10)
        /opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/zclconf/go-cty@v1.0.1-0.20190708163926-19588f92a98f/cty/value_ops.go:620 +0x28c
github.com/hashicorp/terraform/backend/remote.(*Remote).PrepareConfig(0xc00045c6e0, 0x2228c40, 0xc00000e940, 0x0, 0x0, 0x1, 0xc000104ef0, 0x1, 0x1, 0x334a7c0, ...)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/backend/remote/backend.go:146 +0x7c
github.com/hashicorp/terraform/command.(*Meta).backendInitFromConfig(0xc000451860, 0xc0000a6b40, 0x8, 0x44, 0x1, 0xc000000022, 0xc000552068, 0xc00053b1f8, 0x434f2f, 0xc00004e500, ...)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/meta_backend.go:1046 +0x2df
github.com/hashicorp/terraform/command.(*Meta).backend_C_r_s(0xc000451860, 0xc0000a6b40, 0xf5502056, 0xc0004b67e0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/meta_backend.go:669 +0x6f
github.com/hashicorp/terraform/command.(*Meta).backendFromConfig(0xc000451860, 0xc0003f6740, 0x0, 0x0, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/meta_backend.go:519 +0xe3d
github.com/hashicorp/terraform/command.(*Meta).Backend(0xc000451860, 0xc0003f6740, 0x0, 0xc00053b800, 0x1, 0x1, 0x0)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/meta_backend.go:85 +0x5ee
github.com/hashicorp/terraform/command.(*InitCommand).initBackend(0xc000451860, 0xc00017c700, 0x1db149f, 0xf, 0xc0003f6480, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/init.go:472 +0x1c9
github.com/hashicorp/terraform/command.(*InitCommand).Run(0xc000451860, 0xc00003a1a0, 0x1, 0x1, 0xc000074e10)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/init.go:286 +0x1925
github.com/mitchellh/cli.(*CLI).Run(0xc0000d6640, 0xc0000d6640, 0xc00053bd90, 0x1)
        /opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1f1
main.wrappedMain(0x0)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:223 +0xaff
main.realMain(0x0)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:100 +0xb4
main.main()
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:36 +0x3b
```

**Comment by danieldreier** [comment]
I reproduced this issue on 0.12.26 and 0.13.0-beta1

**Comment by mildwonkey** [comment]
This is a duplicate of https://github.com/hashicorp/terraform/issues/23992, so I am going to close it. There is an open PR which will fix this (you can follow #23992 to see when its merged)
